### PR TITLE
Language will be saved in the database of the user during registration 

### PR DESCRIPTION
**Pull request by Tirokk**
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-16T13:14:20Z" title="Wednesday, October 16th 2019, 3:14:20 pm +02:00">Oct 16, 2019</time>_
_Merged <time datetime="2019-10-18T13:11:05Z" title="Friday, October 18th 2019, 3:11:05 pm +02:00">Oct 18, 2019</time>_
---

## 🍰 Pullrequest
set language will be saved in the database of the user during registration 
eingestellte sprache wird beim registrieren in die datenbank zum user gespeichert 

### issuse
#1929 

### group
dieses Problem ist ein Teil von https://github.com/Human-Connection/Human-Connection/issues/1931

FireShot Capture 094 - Human Connection - Human Connection - localhost

FireShot Capture 093 - bolt___localhost_7687 - Neo4j Browser - localhost

### CODE REVIEW

File: backend/src/models/User.js [hunk 1]
```diff
@@ -109,4 +109,8 @@ module.exports = {
     type: 'boolean',
     default: false,
   },
+  locale: {
```
Comment (Tirokk) [inline backend/src/models/User.js:112]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-17T10:48:50Z" title="Thursday, October 17th 2019, 12:48:50 pm +02:00">Oct 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -109,4 +109,8 @@ module.exports = {
     type: 'boolean',
     default: false,
   },
+  language: {
```


If `Vuex-i18n` calls it `locale` maybe we want to call it `locale` in the backend, too?

File: webapp/components/Registration/CreateUserAccount.vue [hunk 2]
```diff
@@ -154,10 +156,19 @@ export default {
...
           mutation: SignupVerificationMutation,
-          variables: { name, password, about, email, nonce, termsAndConditionsAgreedVersion },
+          variables: {
+            name,
+            password,
+            about,
+            email,
+            nonce,
+            termsAndConditionsAgreedVersion,
+            locale,
```
Comment (Tirokk) [inline webapp/components/Registration/CreateUserAccount.vue:170]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-17T10:49:55Z" title="Thursday, October 17th 2019, 12:49:55 pm +02:00">Oct 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -154,10 +156,19 @@ export default {
       const { name, password, about } = this.formData
       const { email, nonce } = this
       const termsAndConditionsAgreedVersion = VERSION
+      const language = this.$i18n.locale()
       try {
         await this.$apollo.mutate({
           mutation: SignupVerificationMutation,
-          variables: { name, password, about, email, nonce, termsAndConditionsAgreedVersion },
+          variables: {
+            name,
+            password,
+            about,
+            email,
+            nonce,
+            termsAndConditionsAgreedVersion,
+            language,
```


Could `LocaleSwitch` component also update the current user's locale?

File: webapp/graphql/Registration.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ export const SignupVerificationMutation = gql`
     $password: String!
     $about: String
     $termsAndConditionsAgreedVersion: String!
+    $locale: String
```
Comment (Tirokk) [inline webapp/graphql/Registration.js:10]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-17T10:50:32Z" title="Thursday, October 17th 2019, 12:50:32 pm +02:00">Oct 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -7,6 +7,7 @@ export const SignupVerificationMutation = gql`
     $password: String!
     $about: String
     $termsAndConditionsAgreedVersion: String!
+    $language: String
```


This is going in the right direction. Could you also add `locale` as allowed input parameter to `UpdateUser`, please?

File: webapp/components/Registration/CreateUserAccount.vue (hunk 2 above)
Comment (Tirokk) [reply webapp/components/Registration/CreateUserAccount.vue:170]: > [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-17T16:33:10Z" title="Thursday, October 17th 2019, 6:33:10 pm +02:00">Oct 17, 2019</time>_

---


ich hatte sie schon in der arbeit. kam aber nicht zu guten ergebnissen. wäre schön wenn ich das im folge PR anpassen könnte wenn es passt und ein besserer weg ist :)

File: webapp/graphql/Registration.js (hunk 3 above)
Comment (Tirokk) [reply webapp/graphql/Registration.js:10]: > [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-17T16:34:14Z" title="Thursday, October 17th 2019, 6:34:14 pm +02:00">Oct 17, 2019</time>_

---


mit dem nächsten PR würde ich das gern + anwendungsfall erledingen um die sache nicht wieder so in die länge zu ziehen

File: backend/src/models/User.js (hunk 1 above)
Comment (Tirokk) [reply backend/src/models/User.js:112]: > [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-17T16:35:15Z" title="Thursday, October 17th 2019, 6:35:15 pm +02:00">Oct 17, 2019</time>_

---


ja

File: webapp/graphql/Registration.js (hunk 3 above)
Comment (Tirokk) [reply webapp/graphql/Registration.js:10]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-17T22:34:56Z" title="Friday, October 18th 2019, 12:34:56 am +02:00">Oct 18, 2019</time>_

---


@ogerly OK, could you rename `language` to `locale`? I'm OK to merge this then.

@mattwr18 what is the better name here? :point_up: I researched `locale` a little and it seems it denominates the language code + country. https://en.wikipedia.org/wiki/Language_code
So it's actually `en-US` and not only `en`. https://en.wikipedia.org/wiki/Locale_(computer_software) We only use the language code. `locale` still good?

Comment (Tirokk) [reply webapp/graphql/Registration.js:10]: > [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-18T10:18:15Z" title="Friday, October 18th 2019, 12:18:15 pm +02:00">Oct 18, 2019</time>_

---


@roschaefer **language** to **_locale_** change

**Comment by Tirokk** [comment]
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-18T13:10:27Z" title="Friday, October 18th 2019, 3:10:27 pm +02:00">Oct 18, 2019</time>_

---

**Review by Mogge — CHANGES_REQUESTED** [review]
Good catch! I would expect that in case I save sth. upon registration, I can always change this at a later time. So, could the `LocaleSwitch` send a backend request in case you are authenticated?

good catch

**Review by Mogge — APPROVED** [review]
Yes! Looks great, feel free to merge it if it passes the build @ogerly

**Pull request closed without merging**